dashboard_server.py

```
@@ -0,0 +1,530 @@
+    def __init__(self, base_dir: str):
+        self.base_dir = Path(base_dir)
+
+    def _load_csv(self, filename: str):
+        path = self.base_dir / filename
+        if not path.exists():
+            return []
+        try:
+            df = pd.read_csv(path)
+            df = df.where(pd.notna(df), None)
+            return df.to_dict('records')
+        except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)